ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -166,6 +167,11 @@ public void run() {
         } catch (Throwable t) {
           LOG.error("Initiator loop caught unexpected exception this time through the loop: " +
               StringUtils.stringifyException(t));
+          if (compactionExecutor != null) {
```

Comment:
same as above, what's the purpose of executor service shutdown? it would be needed only when Initiator thread terminates

Reply:
same as above. This is to fix the number of threads in case threads of executor are lost. This is similarly done in Worker.java too.